src/Microsoft.Azure.WebJobs.Host/Bindings/BindingBase.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+
+    // Helper class for implementing IBinding with the attribute resolver pattern. 
+    internal abstract class BindingBase<TAttribute> : IBinding
+        where TAttribute : Attribute
+    {
+        protected readonly AttributeCloner<TAttribute> _cloner;
+        private readonly ParameterDescriptor _param;
+
+        public BindingBase(AttributeCloner<TAttribute> cloner, ParameterDescriptor param)
+        {
+            _cloner = cloner;
+            _param = param;
+        }
+
+        public BindingBase(AttributeCloner<TAttribute> cloner, ParameterInfo parameterInfo)
+        {
+            _cloner = cloner;
+            _param = new ParameterDescriptor
+            {
+                Name = parameterInfo.Name,
+                DisplayHints = new ParameterDisplayHints
```

Comment:
An extension can provide it's own ParamDescriptor via the other ctor. This ctoris the default case if none is provided. 

---
In reply to: [60988233](https://github.com/Azure/azure-webjobs-sdk/pull/694#discussion_r60988233) [](ancestors = 60988233)